### PR TITLE
refactor(app): add calibration status section and calibration dashboard page

### PR DESCRIPTION
**Pull request by brenthagen**
# Overview

Provides a shell of a calibration dashboard page/route, and a calibration status section on the robot settings calibration tab that links to the calibration dashboard. 

Since the calibration dashboard shares a header design with the Modal component, a `fullPage` flag was added to the ModalShell to render a full page takeover. The `StatusLabel` atom was lightly edited for reuse here and to match the design system.

closes RAUT-190

Screen Shot 2022-09-08 at 12 18 19 PM

Screen Shot 2022-09-08 at 12 19 30 PM

# Changelog

 - Adds robot settings calibration status section and calibration dashboard page shell

# Review requests

 - does full page modal approach make sense?
 - are there any concerns with the changes to StatusLabel?

# Risk assessment

low

### CODE REVIEW

File: app/src/organisms/CalibrationStatusCard/__tests__/CalibrationStatusCard.test.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+describe('CalibrationStatusCard', () => {
+  const props: React.ComponentProps<typeof CalibrationStatusCard> = {
+    robotName: 'otie',
+    setShowDeckCalibrationModal: showDeckCalibrationModal => {},
+  }
+
+  it('renders a calibration status title and description', () => {
+    const [{ getByText }] = render(props)
+
+    getByText('Calibration Status')
```
Comment (koji) [inline app/src/organisms/CalibrationStatusCard/__tests__/CalibrationStatusCard.test.tsx:29]: Seems this case is missing the description part.

```ts
getByText(`For accurate and precise movement, calibrate the robot's deck, pipette offsets, and tip lengths.`)
```

**Review by koji — APPROVED** [review]
Looks good to me!
Just left one commnet.

**Comment by koji** [comment]
For StatusLabel in CalibrationStatusCard, Mel requested us to change the circle size (5px).
https://www.figma.com/file/2xjSLHXquNsKvC1mV9NMjr?node-id=2304:194016#262769975

File: app/src/atoms/StatusLabel/index.tsx [hunk 2]
```diff
@@ -15,18 +17,28 @@ interface StatusLabelProps {
   backgroundColor: string
   iconColor: string
   textColor?: string
+  fontWeight?: number
```
Comment (koji) [inline app/src/atoms/StatusLabel/index.tsx:20]:
```suggestion
  fontWeight?: number
  size?: number
```
https://github.com/Opentrons/opentrons/pull/11446#issuecomment-1242081860

File: app/src/atoms/StatusLabel/index.tsx [hunk 3]
```diff
@@ -51,12 +62,11 @@ export const StatusLabel = (props: StatusLabelProps): JSX.Element | null => {
...
         <StyledText
-          fontSize={TYPOGRAPHY.fontSizeCaption}
+          fontSize={TYPOGRAPHY.fontSizeLabel}
+          fontWeight={fontWeight ?? TYPOGRAPHY.fontWeightRegular}
           color={textColor ?? COLORS.bluePressed}
-          textTransform={TYPOGRAPHY.textTransformCapitalize}
-          marginRight={SPACING.spacing2}
         >
-          {status}
+          {capitalize(status)}
```
Comment (b-cooper) [inline app/src/atoms/StatusLabel/index.tsx:71]: is there a reason JS capitalization is preferable over css text-transform here?

**Review by b-cooper — APPROVED** [review]
💨 🐗   code looks good!

File: app/src/atoms/StatusLabel/index.tsx (hunk 3 above)
Comment (koji) [reply app/src/atoms/StatusLabel/index.tsx:71]: I think Brent is using `capitalize` to capitalize the only first letter.  `text-transform` will make the text `Missing Calibration Data`.
Actually, I thought the exact same thing when I saw the line.

File: app/src/organisms/CalibrationStatusCard/__tests__/CalibrationStatusCard.test.tsx (hunk 1 above)
Comment (brenthagen) [reply app/src/organisms/CalibrationStatusCard/__tests__/CalibrationStatusCard.test.tsx:29]: @koji thanks, I forgot that part

**Pull request merged**